### PR TITLE
NXP-22156: Adapter class and factory with unit tests

### PR DESCRIPTION
**Pull request by bdstech**
Needs peer review and merge

### CODE REVIEW

File: nuxeo-retention-core/pom.xml [hunk 1]
```diff
@@ -1,8 +1,53 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>org.nuxeo.runtime</groupId>
+      <artifactId>nuxeo-runtime-test</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.core</groupId>
+      <artifactId>nuxeo-core-test</artifactId>
+      <version>9.2-SNAPSHOT</version>
```
Comment (akervern) [inline nuxeo-retention-core/pom.xml]: Version is inherited from parent pom. You can remove it

File: nuxeo-retention-core/pom.xml [hunk 2]
```diff
@@ -1,8 +1,53 @@
...
+  ~
+  ~   Contributors:
+  ~       Ryan McCue
+  ~
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
+  <dependencies>
```
Comment (akervern) [inline nuxeo-retention-core/pom.xml]: Should you also move `dependencies` node after the description's one?

File: nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/adapter/Record.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    public static final String RECORD_MAX_RETENTION_AT = "record:max_retention_at";
+
+    protected final DocumentModel doc;
+
+    public Record(DocumentModel doc) {
+        this.doc = doc;
+    }
+
+    // Basic methods
+    //
```
Comment (akervern) [inline nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/adapter/Record.java]: Takes this comments up

File: nuxeo-retention-core/pom.xml (hunk 2 above)
Comment (bdstech) [inline nuxeo-retention-core/pom.xml]: Updated

File: nuxeo-retention-core/pom.xml (hunk 1 above)
Comment (bdstech) [inline nuxeo-retention-core/pom.xml]: updated

File: nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/adapter/Record.java (hunk 3 above)
Comment (bdstech) [inline nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/adapter/Record.java]: updated

**Pull request merged**